hal/inc/hal_platform_compat.h

```diff
@@ -87,4 +87,6 @@
 #define HAL_PLATFORM_NETWORK_MULTICAST (1)
 #endif // HAL_PLATFORM_WIFI
 
+#define HAL_PLATFORM_BACKUP_RAM (1)
```

Comment:
Core build seems to be broken. We are anyway removing it, but let's for now move this define above under `PLATFORM_ID >= PLATFORM_PHOTON_PRODUCTION && PLATFORM_ID != PLATFORM_NEWHAL` check.